src/mac/streams/thread_names.rs

```diff
@@ -0,0 +1,91 @@
+use super::*;
+
+impl MinidumpWriter {
+    /// Writes the [`MDStreamType::ThreadNamesStream`] which is an array of
+    /// [`miniduimp_common::format::MINIDUMP_THREAD`]
+    pub(crate) fn write_thread_names(
+        &mut self,
+        buffer: &mut DumpBuf,
+        dumper: &TaskDumper,
+    ) -> Result<MDRawDirectory, WriterError> {
+        let threads = dumper.read_threads()?;
+
+        // Ignore the thread that handled the exception
+        let thread_count = if self.crash_context.handler_thread != mach2::port::MACH_PORT_NULL {
+            threads.len() - 1
+        } else {
+            threads.len()
+        };
```

Comment:
It'd be nice to put this in a helper function with a name that describes the intent so we can share it with `MinidumpWriter::write_thread_list()` and drop the comment.